### ISSUE TITLE
Exception and CasparCG 2.2.0 doesn't start. 

### ISSUE THREAD

**Issue by alexdjca** [CasparCG/server#1180]
Hi All,

No matter how i configure the server, there is no way to start it in my machine.

The problem happens only on 2.2.0, but 2.0.7 is not affected.

Here are the logs and thanks in advance to who can help me

Regards,
Alex

```
[2019-02-19 13:36:28.726] [info]    ############################################################################
[2019-02-19 13:36:28.727] [info]    CasparCG Server is distributed by the Swedish Broadcasting Corporation (SVT)
[2019-02-19 13:36:28.727] [info]    under the GNU General Public License GPLv3 or higher.
[2019-02-19 13:36:28.727] [info]    Please see LICENSE.TXT for details.
[2019-02-19 13:36:28.727] [info]    http://www.casparcg.com/
[2019-02-19 13:36:28.727] [info]    ############################################################################
[2019-02-19 13:36:28.727] [info]    Starting CasparCG Video and Graphics Playout Server 2.2.0 66a9e3e2 Stable
[2019-02-19 13:36:28.730] [info]    Initialized ffmpeg module.
[2019-02-19 13:36:28.730] [info]    Initialized oal module.
[2019-02-19 13:36:28.731] [info]    Initialized decklink module.
[2019-02-19 13:36:28.731] [info]    Initialized screen module.
[2019-02-19 13:36:28.770] [info]    Initialized html module.
[2019-02-19 13:36:28.817] [info]    Initialized flash module.
[2019-02-19 13:36:28.817] [info]    Initialized newtek module.
[2019-02-19 13:36:28.817] [info]    Initialized bluefish module.
[2019-02-19 13:36:28.817] [info]    Initialized image module.
[2019-02-19 13:36:28.818] [info]    "C:/Users\Administrator\Desktop\casparcg-220\casparcg.config":
[2019-02-19 13:36:28.818] [info]    -----------------------------------------
[2019-02-19 13:36:28.818] [info]    <?xml version="1.0" encoding="utf-8"?>
[2019-02-19 13:36:28.818] [info]    <configuration>
[2019-02-19 13:36:28.818] [info]       <log-level>trace [trace|debug|info|warning|error|fatal]</log-level>
[2019-02-19 13:36:28.818] [info]       <paths>
[2019-02-19 13:36:28.818] [info]          <media-path>media/</media-path>
[2019-02-19 13:36:28.818] [info]          <log-path>log/</log-path>
[2019-02-19 13:36:28.818] [info]          <data-path>data/</data-path>
[2019-02-19 13:36:28.818] [info]          <template-path>template/</template-path>
[2019-02-19 13:36:28.818] [info]          <font-path>font/</font-path>
[2019-02-19 13:36:28.818] [info]       </paths>
[2019-02-19 13:36:28.818] [info]       <lock-clear-phrase>secret</lock-clear-phrase>
[2019-02-19 13:36:28.818] [info]       <channels>
[2019-02-19 13:36:28.818] [info]          <channel>
[2019-02-19 13:36:28.818] [info]             <video-mode>PAL</video-mode>
[2019-02-19 13:36:28.818] [info]             <consumers>
[2019-02-19 13:36:28.818] [info]                <screen/>
[2019-02-19 13:36:28.818] [info]                <ffmpeg>
[2019-02-19 13:36:28.818] [info]                   <path>udp://239.123.234.11:11111?pkt_size=1316</path>
[2019-02-19 13:36:28.818] [info]                   <args>-filter:v scale=720:576,format=yuv420p -g:v 50 -codec:v libx264 -format mpegts</args>
[2019-02-19 13:36:28.818] [info]                </ffmpeg>
[2019-02-19 13:36:28.818] [info]             </consumers>
[2019-02-19 13:36:28.818] [info]          </channel>
[2019-02-19 13:36:28.818] [info]       </channels>
[2019-02-19 13:36:28.818] [info]       <controllers>
[2019-02-19 13:36:28.818] [info]          <tcp>
[2019-02-19 13:36:28.818] [info]             <port>5250</port>
[2019-02-19 13:36:28.818] [info]             <protocol>AMCP</protocol>
[2019-02-19 13:36:28.818] [info]          </tcp>
[2019-02-19 13:36:28.818] [info]       </controllers>
[2019-02-19 13:36:28.818] [info]       <amcp>
[2019-02-19 13:36:28.818] [info]          <media-server>
[2019-02-19 13:36:28.818] [info]             <host>localhost</host>
[2019-02-19 13:36:28.818] [info]             <port>8000</port>
[2019-02-19 13:36:28.818] [info]          </media-server>
[2019-02-19 13:36:28.818] [info]       </amcp>
[2019-02-19 13:36:28.818] [info]    </configuration>
[2019-02-19 13:36:28.818] [info]    -----------------------------------------
[2019-02-19 13:36:29.244] [info]    Initializing OpenGL Device.
[2019-02-19 13:36:29.245] [info]    Initialized OpenGL 4.5.0 NVIDIA 418.81 NVIDIA Corporation
[2019-02-19 13:36:29.275] [info]    Initialized OpenGL Accelerated GPU Image Mixer for channel 1
[2019-02-19 13:36:29.276] [info]    video_channel[1|PAL] Successfully Initialized.
[2019-02-19 13:36:29.277] [info]    Screen consumer [1|PAL] Initialized.
[2019-02-19 13:36:29.278] [info]    ffmpeg[udp://239.123.234.11:11111?pkt_size=1316] Initialized.
[2019-02-19 13:36:29.278] [info]    Initialized channels.
[2019-02-19 13:36:29.282] [info]    [asio] Shutting down global io_service.
[2019-02-19 13:36:29.282] [info]    [asio] Global io_service uninitialized.
[2019-02-19 13:36:29.283] [info]    video_channel[1|PAL] Uninitializing.
[2019-02-19 13:36:29.283] [info]    [ffmpeg] [libx264 @ 000000000231B280] using SAR=64/45
[2019-02-19 13:36:29.283] [info]    
[2019-02-19 13:36:29.284] [info]    [ffmpeg] [libx264 @ 000000000231B280] using cpu capabilities: MMX2 SSE2Fast SSSE3 SSE4.2
[2019-02-19 13:36:29.284] [info]    
[2019-02-19 13:36:29.292] [info]    [ffmpeg] [libx264 @ 000000000231B280] profile High, level 3.1
[2019-02-19 13:36:29.292] [info]    
[2019-02-19 13:36:29.333] [info]    ffmpeg[udp://239.123.234.11:11111?pkt_size=1316] Uninitialized.
[2019-02-19 13:36:29.408] [info]    Screen consumer [1|PAL] Uninitialized.
[2019-02-19 13:36:29.429] [error]   Exception: Throw location unknown (consider using BOOST_THROW_EXCEPTION)
[2019-02-19 13:36:29.429] [error]   Dynamic exception type: class boost::exception_detail::clone_impl<struct boost::exception_detail::error_info_injector<class boost::system::system_error> >
[2019-02-19 13:36:29.429] [error]   std::exception::what: bind: An attempt was made to access a socket in a way forbidden by its access permissions
[2019-02-19 13:36:29.429] [error]   

[2019-02-19 13:36:29.429] [error]    0# 0x000000013F6E98AE in casparcg
[2019-02-19 13:36:29.429] [error]    1# 0x000000013F6E91DF in casparcg
[2019-02-19 13:36:29.429] [error]    2# 0x000000013FB81CE7 in casparcg
[2019-02-19 13:36:29.429] [error]    3# _C_specific_handler in VCRUNTIME140
[2019-02-19 13:36:29.429] [error]    4# _TypeMatch in VCRUNTIME140
[2019-02-19 13:36:29.429] [error]    5# RtlRestoreContext in ntdll
[2019-02-19 13:36:29.429] [error]    6# 0x000000013F6F077C in casparcg
[2019-02-19 13:36:29.429] [error]    7# 0x000000013FADC638 in casparcg
[2019-02-19 13:36:29.429] [error]    8# BaseThreadInitThunk in kernel32
[2019-02-19 13:36:29.429] [error]    9# RtlUserThreadStart in ntdll
[2019-02-19 13:36:29.429] [error]   
[2019-02-19 13:36:29.455] [fatal]   Unhandled exception in main thread. Please report this error on the GitHub project page (www.github.com/casparcg/server/issues).
```

**Comment by Julusian** [comment]
Do you have another instance of caspar running? 
It looks like the AMCP port is already in use: `bind: An attempt was made to access a socket in a way forbidden by its access permissions`

**Comment by Julusian** [comment]
I managed to reproduce this recently, and it was due to the AMCP port being in use by another program.
While that error could be caught and guarded against, is that really wanted? Doing so would mean that caspar is running without listening to AMCP and so would be rather useless.

When I had this occur, windows tells me it was the client that was using the port:
![image (1)](https://user-images.githubusercontent.com/1327476/77306695-f1ec4c00-6cef-11ea-9b8f-06287b8082c5.png)
